numcodecs/_fletcher.c

```diff
@@ -0,0 +1,43 @@
+#include <stdint.h>
+#include <stddef.h>
+
+// https://github.com/Unidata/netcdf-c/blob/8eb71290eb9360dcfd4955ba94759ba8d02c40a9/plugins/H5checksum.c
+
+
+uint32_t H5_checksum_fletcher32(const void *_data, size_t _len)
```

Comment:
Can we please move this to Cython? We've dropped all C code from this repo

Reply:
This is not generated C code, though, and cythonising something that already is C is both painful and inefficient

Reply:
It's really not. For example [this header was in C]( https://github.com/zarr-developers/numcodecs/blob/cde8d624c5516854a5498df0be7fb1c7c5e766ef/numcodecs/stdint_compat.h ) and we moved it to [Cython]( https://github.com/zarr-developers/numcodecs/blob/955019b55a31b4a03c76f62edfde5df2301282c4/numcodecs/_utils.pxd ). This also stripped out a bunch of boilerplate that was no longer needed.